src/index.js

```diff
@@ -1,45 +1,24 @@
-#!/usr/bin/env node
-
-import yargs from 'yargs';
 import metarpheusTcomb from './metarpheusTcomb';
 import IntermRep from './IntermRep';
 import Config from './Config';
-import fs from 'fs';
-import path from 'path';
-
-const argv = yargs.argv;
-const configFileArg = 'config';
-if (!argv[configFileArg]) {
-  throw new Error(`missing --${configFileArg}=path/to/config.js[on]`);
-}
-
-require('babel/register')({
-  only: new RegExp(argv[configFileArg].replace(/\.js$/, '')), // so that config file can be ESwhatever
-  extensions: ['.js'],
-  stage: 0,
-  loose: true
-});
-
-const {
-  intermRepIn,
-  modelPrelude = `// DO NOT EDIT MANUALLY - metarpheus-generated
 
-`,
-  apiPrelude = `// DO NOT EDIT MANUALLY - metarpheus-generated
+export default function({ intermRep: _intermRep, config }) {
+  const {
+    intermRepIn,
+    modelPrelude = `// DO NOT EDIT MANUALLY - metarpheus-generated
 
-`,
-  apiModelPrefix = '',
-  modelOut = 'model.js',
-  apiOut = '',
-  overrides = {},
-  renameModel = v => v
-} = Config(require(path.resolve(process.cwd(), argv[configFileArg])));
+  `,
+    apiPrelude = `// DO NOT EDIT MANUALLY - metarpheus-generated
 
-const intermRep = IntermRep(require(path.resolve(process.cwd(), intermRepIn)));
+  `,
+    apiModelPrefix = '',
+    modelOut = 'model.js',
+    apiOut = '',
+    overrides = {},
+    renameModel = v => v
+  } = Config(config);
```

Comment:
discussed offline, `interface` is just an identity as every other tcomb fn. `struct` is the only one "creating instances"